src/app/view/vessel-editor/vessel-editor.component.html

```diff
@@ -1,72 +1,131 @@
         </span>
       </div>
     </div>
+
     <div class="p-field grid">
-      <label for="vesselOperatorCarrierCode" class="col-12 p-mb-2 md:col-2 p-mb-md-0">{{ 'general.vessel.vesselOperatorCarrierCode' | translate }}</label>
+      <label for="length" class="col-12 p-mb-2 md:col-2 p-mb-md-0">{{ 'general.vessel.length.name' | translate
+        }}</label>
       <div class="col-12 md:col-10">
-      <p-dropdown appendTo="body" [(ngModel)]="vessel.vesselOperatorCarrierCode" [filter]="true" [required] = "true" [options]="carriers"  [ngModelOptions]="{standalone: true}"
-      placeholder="{{ 'general.carrier.select' | translate }}"></p-dropdown>
+        <input id="length" type="text" formControlName="length" pInputText>
+        <span *ngIf="vesselFormGroup.get('length').hasError('pattern')" class="p-invalid">
+          {{ 'general.vessel.length.pattern' | translate }}
+        </span>
+        <span [hidden]="hasDimensionUnit()" class="p-warning">
+          {{ 'general.vessel.length.missingUnit' | translate }}
+        </span>
+      </div>
+    </div>
+
+    <div class="p-field grid">
+      <label for="width" class="col-12 p-mb-2 md:col-2 p-mb-md-0">{{ 'general.vessel.width.name' | translate }}</label>
+      <div class="col-12 md:col-10">
+        <input id="width" type="text" formControlName="width" pInputText>
+        <span *ngIf="vesselFormGroup.get('width').hasError('pattern')" class="p-invalid">
+          {{ 'general.vessel.width.pattern' | translate }}
+        </span>
+        <span [hidden]="hasDimensionUnit()" class="p-warning">
+          {{ 'general.vessel.width.missingUnit' | translate }}
+        </span>
+      </div>
+    </div>
+
+    <div class="p-field grid">
+      <label for="vesselOperatorCarrierCode" class="col-12 p-mb-2 md:col-2 p-mb-md-0">{{
+        'general.vessel.vesselOperatorCarrierCode' | translate }}</label>
+      <div class="col-12 md:col-10">
+        <p-dropdown appendTo="body" [filter]="true" [options]="carriers" formControlName="vesselOperatorCarrierCode"
+          placeholder="{{ 'general.carrier.select' | translate }}"></p-dropdown>
+      </div>
+    </div>
+
+    <div class="p-field grid">
+      <label for="vesselType" class="col-12 p-mb-2 md:col-2 p-mb-md-0">{{ 'general.vessel.vesselType.name' | translate
+        }}</label>
+      <div class="col-12 md:col-10">
+        <p-dropdown appendTo="body" [filter]="true" formControlName="type" [options]="vesselTypes"></p-dropdown>
+      </div>
+    </div>
+
+    <div class="p-field grid">
+      <label for="dimensionUnit" class="col-12 p-mb-2 md:col-2 p-mb-md-0">{{ 'general.vessel.dimensionUnit.name' |
+        translate }}</label>
+      <div class="col-12 md:col-10">
+        <p-dropdown appendTo="body" [filter]="true" formControlName="dimensionUnit" [options]="dimensionUnits"
+          (onChange)="toggleDimensionFields()">
+        </p-dropdown>
       </div>
     </div>
```

Comment:
Can we make the dimensionUnit field editable only if it is not set originally?

(asking because the backend does not handle changing the dimension unit correctly retroactively)

Reply:
Yes, should be possible. I will implement that as a new commit.